auth_supplier/controllers/main.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# (c) 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp.addons.auth_signup.controllers.main import AuthSignupHome
+from openerp.http import request
+
+
+class AuthSignupHome(AuthSignupHome):
+
+    def do_signup(self, qcontext):
+        """ Shared helper that creates a res.partner out of a token """
+        values = dict(
+            (key, qcontext.get(key)) for key in (
+                'login', 'name', 'password', 'account_type'))
+        assert any([k for k in values.values()]),\
```

Comment:
[Assert should not be used for production](https://docs.python.org/2.7/reference/simple_stmts.html#assert), where optimization is quite common.